Azure/sonic-mgmt
# [sanity_check]: Improve mux simulator sanity check

Signed-off-by: Lawrence Lee <lawlee@microsoft.com>


### Description of PR


Summary:
Fixes https://github.com/Azure/sonic-buildimage/issues/8105

### Type of change





### Back port request

### Approach
#### What is the motivation for this PR?
Current mux simulator sanity check is very flaky since it is not able to directly send/receive packets on the DUT, instead it runs commands on the DUT that should generate specific packets.

#### How did you do it?
- Create fixture to get status of all mux simulator ports from server
- Change mux sim sanity check to examine returned status/flows instead
of sending traffic
- Add support to sanity check for infrastructure recovery actions

#### How did you verify/test it?
- Run the sanity check on a healthy dual ToR testbed, verify passing
- Manually remove some mux simulator flows
    - Run the sanity check
    - Verify the test fails, and for the correct reason
    - Verify the mux simulator recovery action is run
    - Verify the second sanity check passes

#### Any platform specific information?

#### Supported testbed topology if it's a new test case?

### Documentation
